### PR TITLE
fix(cli): handle Windows paths (#2312)

### PR DESCRIPTION
**Pull request by bburns**
This commit uses forward slashes on Windows for the calls to glob.sync and tsup.build - those are where `nango compile` and `nango dev` were failing.

I copied the slash function from https://github.com/sindresorhus/slash, as it just converts \ to /, and didn't want to deal with package-lock.json changes.

It also uses path.join or path.resolve instead of `${foo}/pok/bar.ts`, os.tmpdir() instead of '/tmp', etc.

getNangoRootPath() now always returns a string, since getPackagePath() will throw an error if it can't find the package.

And I expanded the regex in packages/node-client/lib/utils.unit.test.ts to handle "nango-node-client/0.41.0 (linux/5.15.153.1-microsoft-standard-WSL2; node.js/20.12.2)".

I ran the tests on Windows with git bash and Command Prompt, and with WSL2 Ubuntu -

    npm i
    npm run ts-build
    npm run test:cli
    npm run test:unit

and in a test nango project, these worked -

    $ node /c/users/bburns/Workspace/forks/nango/packages/cli/dist/index.js generate --debug
    $ node /c/users/bburns/Workspace/forks/nango/packages/cli/dist/index.js compile --debug
    $ node /c/users/bburns/Workspace/forks/nango/packages/cli/dist/index.js dev --debug

Also added a workflow to run test:cli and test:unit on Windows - .github/workflows/tests-windows.yaml.

## Issue ticket number and link

#2312

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [x] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

**Comment by bodinsamuel** [comment]
Thanks for your contribution, it was on our todo list so definitely helps. Reviewing that rn 👍🏻

File: packages/cli/lib/tests/helpers.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import path from 'node:path';
 import fs from 'node:fs/promises';
+import os from 'node:os';
 
-export const fixturesPath = path.join(__dirname, '..', '..', 'fixtures');
+export const fixturesPath = './packages/cli/fixtures';
```
Comment (bodinsamuel) [inline packages/cli/lib/tests/helpers.ts:5]: I remember having issues with this, but seems to work fine 🤔 I should have added a comment

File: packages/cli/lib/utils.ts [hunk 2]
```diff
@@ -271,23 +271,18 @@ export function getUserAgent(): string {
...
 
-export function getNangoRootPath(debug = false) {
+export function getNangoRootPath(debug = false): string {
     const packagePath = getPackagePath(debug);
-    if (!packagePath) {
-        if (debug) {
-            printDebug('Could not find nango cli root path locally');
-        }
-        return null;
-    }
```
Comment (bodinsamuel) [inline packages/cli/lib/utils.ts:281]: Indeed legacy stuff, thanks

File: .github/workflows/tests-windows.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+concurrency:
+    group: tests-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    tests:
+        runs-on: windows-latest
+        strategy:
+            matrix:
+                node-version: [18.x, 20.x]
```
Comment (bodinsamuel) [inline .github/workflows/tests-windows.yaml]: I think it's possible to use the existing `tests.yaml` and add the os in the matrix 

example https://github.com/vitest-dev/vitest/blob/main/.github/workflows/ci.yml

**Review by bodinsamuel — COMMENTED** [review]
Massive work 💪🏻 
Minor comments, but looks good

**Comment by bburns** [comment]
Okay, I wasn't sure if test:integration should run on Windows also, so
separated the others out.

**Comment by bburns** [comment]
I realized I needed to set the tests-windows.yaml concurrency group to something unique, so fixed that, rebased to the latest upstream, and tested locally. 

It looks okay, except for the deployment actions - maybe could add some guards for if the PR is coming from a fork? I'm not too familiar with github ci/cd yet...

File: packages/cli/lib/tests/helpers.ts (hunk 1 above)
Comment (bburns) [reply packages/cli/lib/tests/helpers.ts:5]: yeah __dirname seemed to be missing in that file...

**Review by bodinsamuel — APPROVED** [review]
Looks good! 💪🏻 
I just need to discuss one point with the team otherwise it's good to merge

**Comment by bodinsamuel** [comment]
Fixes https://linear.app/nango/issue/NAN-1324/nango-cli-doesnt-compile-on-windows

**Comment by bodinsamuel** [comment]
Alright, I just modified the windows tests to only run CLI, because there are other issues to compile to full codebase in Windows.

**Review by bodinsamuel — APPROVED** [review]
Thanks a lot for this deeply needed change 🙏🏻

**Comment by bburns** [comment]
No problem - love this project!

**Comment by bodinsamuel** [comment]
It has been released in `0.42.0` 🙌🏻

**Pull request merged**